core/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1447,6 +1429,34 @@ public QueuedMessage(ProtocolMessage msg, CompletionListener listener) {
         }
     }
 
+    /**
+     * Send all pending messages which are queue.
+     * Remove them from the queue once they are send successfully
+     * Spec: RTN19a
+     */
+    public void sendPendingQueueMessages() {
+        //RTN19a
+        if (pendingMessages.queue != null && !pendingMessages.queue.isEmpty()) {
+            for (final QueuedMessage queuedMessage : pendingMessages.queue) {
+                try {
+                    send(queuedMessage.msg, false, new CompletionListener() {
+                        @Override
+                        public void onSuccess() {
+                            pendingMessages.queue.remove(queuedMessage);
```

Comment:
I don't understand what you're doing here here. Why do we have a completion listener here? send() will add the message into the pending message list, so it'll now have both instances in there? `pendingMessages#ack()` and `nack()` will handle acks and nacks, so now they'll be handled twice?

Reply:
I thought we should remove a message from the pending list once it is successfully sent. This way we won't have that same message in a queue state as I do not see anywhere that items are removed from the queue.

Reply:
Here is the solution we agreed on the meeting https://github.com/ably/ably-java/pull/842/commits/5fae005bdb9ace00c18099a147e315be582ab00c